From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: quh4gko8 <88831734+quh4gko8@users.noreply.github.com>
Date: Wed, 26 Feb 2025 09:46:02 +0000
Subject: [PATCH] Always allow platform CredentialManager calls for Android 15+

---
diff --git a/components/webauthn/android/java/src/org/chromium/components/webauthn/AuthenticatorImpl.java b/components/webauthn/android/java/src/org/chromium/components/webauthn/AuthenticatorImpl.java
index 20efaf8955a0..39ac05ecd4e8 100644
--- a/components/webauthn/android/java/src/org/chromium/components/webauthn/AuthenticatorImpl.java
+++ b/components/webauthn/android/java/src/org/chromium/components/webauthn/AuthenticatorImpl.java
@@ -14,6 +14,7 @@ import android.annotation.SuppressLint;
 import android.app.PendingIntent;
 import android.content.Context;
 import android.content.Intent;
+import android.os.Build;
 import android.os.Bundle;
 import android.util.Pair;
 
@@ -167,8 +168,10 @@ public final class AuthenticatorImpl implements Authenticator, AuthenticationCon
         mIsPaymentRequest = options.isPaymentCredentialCreation;
         mRequestCallback = requestCallback;
         mRequestCallback.setCompletionCallback(this::cleanupRequest);
-        if (!GmsCoreUtils.isWebauthnSupported()
-                || (!isChrome(mWebContents) && !GmsCoreUtils.isResultReceiverSupported())) {
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.VANILLA_ICE_CREAM
+                && (!GmsCoreUtils.isWebauthnSupported()
+                        || (!isChrome(mWebContents)
+                                && !GmsCoreUtils.isResultReceiverSupported()))) {
             mRequestCallback.onComplete(
                     WebauthnRequestResponse.forFailedMakeCredential(
                             AuthenticatorStatus.NOT_IMPLEMENTED,
@@ -254,9 +257,11 @@ public final class AuthenticatorImpl implements Authenticator, AuthenticationCon
                        && DeviceFeatureMap.isEnabled(
                                DeviceFeatureList.WEBAUTHN_IMMEDIATE_GET);
-        if (!GmsCoreUtils.isWebauthnSupported()
-                || (!isChrome(mWebContents) && !GmsCoreUtils.isResultReceiverSupported())
-                || (options.publicKey == null && !isPasswordOnlyFlux)) {
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.VANILLA_ICE_CREAM
+                && (!GmsCoreUtils.isWebauthnSupported()
+                        || (!isChrome(mWebContents)
+                                && !GmsCoreUtils.isResultReceiverSupported())
+                        || (options.publicKey == null && !isPasswordOnlyFlux))) {
             RequestMetrics metrics =
                     new RequestMetrics.Builder()
                             .setGetAssertionOutcome(GetAssertionOutcome.OTHER_FAILURE)
@@ -300,10 +305,16 @@ public final class AuthenticatorImpl implements Authenticator, AuthenticationCon
     }
 
     private boolean couldSupportConditionalMediation() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.VANILLA_ICE_CREAM) {
+            return true;
+        }
         return GmsCoreUtils.isWebauthnSupported() && isChrome(mWebContents);
     }
 
     private boolean couldSupportUvpaa() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.VANILLA_ICE_CREAM) {
+            return true;
+        }
         return GmsCoreUtils.isWebauthnSupported()
                 && (isChrome(mWebContents) || GmsCoreUtils.isResultReceiverSupported());
     }
diff --git a/components/webauthn/android/java/src/org/chromium/components/webauthn/cred_man/CredManSupportProvider.java b/components/webauthn/android/java/src/org/chromium/components/webauthn/cred_man/CredManSupportProvider.java
index 26fa7d1c6b43..8bc2e71f3d1c 100644
--- a/components/webauthn/android/java/src/org/chromium/components/webauthn/cred_man/CredManSupportProvider.java
+++ b/components/webauthn/android/java/src/org/chromium/components/webauthn/cred_man/CredManSupportProvider.java
@@ -82,7 +82,9 @@ public class CredManSupportProvider {
             log(TAG, "Disabled because of Android version.");
             return sCredManSupport;
         }
-        if (notSkippedBecauseInTests() && hasOldGmsVersion()) {
+        if (getAndroidVersion() < Build.VERSION_CODES.VANILLA_ICE_CREAM
+                && notSkippedBecauseInTests()
+                && hasOldGmsVersion()) {
             sCredManSupport = CredManSupport.DISABLED;
             log(TAG, "Disabled because of old GMS version.");
             return sCredManSupport;
